homeassistant/components/voice_assistant/pipeline.py

```diff
@@ -0,0 +1,126 @@
+"""Classes for voice assistant pipelines."""
+from __future__ import annotations
+
+import asyncio
+from collections.abc import Callable
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Any
+
+from homeassistant.components import conversation
+from homeassistant.core import Context, HomeAssistant
+from homeassistant.util.dt import utcnow
+
+DEFAULT_TIMEOUT = 30  # seconds
+
+
+@dataclass
+class PipelineRequest:
+    """Request to start a pipeline run."""
+
+    intent_input: str | None = None
+    conversation_id: str | None = None
+
+
+class PipelineEventType(str, Enum):
```

Comment:
https://docs.python.org/3/library/enum.html#enum.StrEnum

```suggestion
class PipelineEventType(StrEnum):
```

Reply:
Looks like it's new in 3.11 :(

Reply:
`from homeassistant.backports.enum import StrEnum`